spec/marc_to_solr/lib/solr_deleter_spec.rb

```diff
@@ -44,22 +44,21 @@
       end
     end
 
-    context 'when the HTTP request to the Solr endpoint' do
+    context 'when the HTTP request to Solr times out' do
       let(:logger) do
         instance_double(ActiveSupport::Logger)
       end
 
       before do
-        allow(Faraday).to receive(:post).and_raise(Net::ReadTimeout)
+        allow(Faraday).to receive(:post).and_raise(Faraday::TimeoutError)
         allow(logger).to receive(:info)
         allow(logger).to receive(:warn)
-        allow(Rails).to receive(:logger).and_return(logger)
 
         solr_deleter.delete(ids, batch_size)
       end
 
       it 'logs a warning' do
-        expect(logger).to have_received(:warn).exactly(2).times.with("Failed to transmit the POST request to #{url}: <delete><id>#{ids.first}</id><id>#{ids.last}</id></delete>")
+        expect(logger).to have_received(:warn).exactly(2).times
```

Comment:
I could not figure out how to write a `expect` that evaluates if the message includes a text (e.g. "time out") which is required now that I am including the elapsed time in the message.

Reply:
I'm pretty sure you can pass it a regex instead of a string and then it will test there's a match on the regex.

Reply:
example: https://github.com/pulibrary/bibdata/blob/c642e47575620b4f2a2f9e391f302d9bd9f219f0/spec/services/numismatics_indexer_spec.rb#L78

Reply:
@hackartisan Great idea, thanks!